### ISSUE TITLE
Edition.get returns 'empty' edition which can be saved

### ISSUE THREAD

**Issue by cdrini** [internetarchive/openlibrary-client#71]
When `Edition.get` has no results, it returns an 'empty' edition with an error field on it. This edition can be (and was) saved, and is now on OpenLibrary: https://openlibrary.org/books/None

Input:
`ol.Edition.get(isbn='foobar')`
Output:
```
<class 'olclient.openlibrary.OpenLibrary.Edition.<locals>.Edition'
{'_work': None,
'work_olid': None,
'olid': 'None',
'description': None,
'notes': None,
'identifiers': {'wikidata': ['Q2071569']},
'title': None,
'subtitle': None,
'pages': None,
'authors': [],
'publisher': None,
'publish_date': None,
'cover': None,
'created': {'type': '/type/datetime', 'value': '2018-02-25T22:48:09.562592'},
'last_modified': {'type': '/type/datetime', 'value': '2018-02-25T22:48:09.562592'},
'latest_revision': 1,
'error': 'notfound',
'type': {'key': '/type/edition'},
'revision': 1}>
```

**Comment by cdrini** [comment]
Similarly with works:

```py
ol.Work.get('foobar')
```
```
<class 'olclient.openlibrary.OpenLibrary.Work.<locals>.Work' {
'olid': 'foobar',
'_editions': [],
'description': None,
'notes': None,
'key': '/works/foobar',
'error': 'notfound'}>
```

**Comment by hornc** [comment]
same as #60

**Comment by hornc** [comment]
Oh dear, I thought I had fixed this with PR #106,
but there are still problems
```
>>> edition = ol.Edition.get_olid_by_isbn('foobar')
>>> edition is None
True

```

and that None is used to fetch a url,

```
>>> url = ol.base_url + '/books/%s.json' % None
>>> url
u'https://openlibrary.org/books/None.json'
```

Which is now a real item with metadata... so it treats it as a real item. https://openlibrary.org/books/None.json
